sdk/textanalytics/Azure.AI.TextAnalytics/src/TextAnalyticsClient.cs

```diff
@@ -2072,6 +2072,451 @@ private Response<RecognizeLinkedEntitiesResultCollection> RecognizeLinkedEntitie
+            scope.Start();
+
+            try
+            {
+                ResponseWithHeaders<TextAnalyticsHealthHeaders> response = await _serviceRestClient.HealthAsync(documentInputs, options.ModelVersion, _stringCodeUnit, cancellationToken).ConfigureAwait(false);
+                string location = response.Headers.OperationLocation;
+
+                return new AnalyzeHealthOperation(_serviceRestClient, _clientDiagnostics, location, documentInputs);
+            }
+            catch (Exception e)
+            {
+                scope.Failed(e);
+                throw;
+            }
+        }
+
+        private AnalyzeHealthOperation StartAnalyzeHealth(MultiLanguageBatchInput batchInput, AnalyzeHealthOptions options, CancellationToken cancellationToken = default)
+        {
+            options ??= new AnalyzeHealthOptions();
+
+            using DiagnosticScope scope = _clientDiagnostics.CreateScope($"{nameof(TextAnalyticsClient)}.{nameof(StartAnalyzeHealth)}");
+            scope.Start();
+
+            try
+            {
+                ResponseWithHeaders<TextAnalyticsHealthHeaders> response = _serviceRestClient.Health(batchInput, options.ModelVersion, _stringCodeUnit, cancellationToken);
+                string location = response.Headers.OperationLocation;
+
+                return new AnalyzeHealthOperation(_serviceRestClient, _clientDiagnostics, location, batchInput);
+            }
+            catch (Exception e)
+            {
+                scope.Failed(e);
+                throw;
+            }
+        }
+
+        private async Task<AnalyzeHealthOperation> StartAnalyzeHealthAsync(MultiLanguageBatchInput batchInput, AnalyzeHealthOptions options, CancellationToken cancellationToken = default)
+        {
+            options ??= new AnalyzeHealthOptions();
+
+            using DiagnosticScope scope = _clientDiagnostics.CreateScope($"{nameof(TextAnalyticsClient)}.{nameof(StartAnalyzeHealth)}");
+            scope.Start();
+
+            try
+            {
+                ResponseWithHeaders<TextAnalyticsHealthHeaders> response = await _serviceRestClient.HealthAsync(batchInput, options.ModelVersion, _stringCodeUnit, cancellationToken).ConfigureAwait(false);
+                string location = response.Headers.OperationLocation;
+
+                return new AnalyzeHealthOperation(_serviceRestClient, _clientDiagnostics, location, batchInput, options.Top, options.Skip, options.ShowStats);
+            }
+            catch (Exception e)
+            {
+                scope.Failed(e);
+                throw;
+            }
+        }
+
+        /// <summary> Cancel healthcare prediction job. </summary>
+        /// <param name="operation"> operation. </param>
+        /// <param name="cancellationToken"> The cancellation token to use. </param>
+        // Look for other alternatives of operation cancellation
+        #pragma warning disable AZC0003 // DO make service methods virtual.
```

Comment:
any reason why this was disable?

Reply:
I am returning Operation Location id for which it complains as it is a string. Though cancellation call is returning a header which is internal and making it public is causing various more errors.

Reply:
this warning is complaining because method was not virtual. I see you changed it so the suppression shouldn't' be needed more. right?

Reply:
removed